cluster-operator/src/test/java/io/strimzi/operator/cluster/model/VolumeUtilsTest.java

```diff
@@ -227,4 +230,57 @@ public void testPodSetVolumesEphemeralOnly()    {
         assertThat(volumes.get(0).getName(), is("data"));
         assertThat(volumes.get(0).getEmptyDir(), is(notNullValue()));
     }
+
+    @ParallelTest
+    public void testKRaftMetadataPath() {
+        // Test with unmarked no-JBOD storage
+        assertThat(VolumeUtils.kraftMetadataPath(PERSISTENT_CLAIM_STORAGE), is("/var/lib/kafka/data"));
+        assertThat(VolumeUtils.kraftMetadataPath(EPHEMERAL_STORAGE), is("/var/lib/kafka/data"));
+
+        // Test with marked no-JBOD storage
+        PersistentClaimStorage markedSingleVolume = new PersistentClaimStorageBuilder()
+                .withDeleteClaim(false)
+                .withSize("20Gi")
+                .withKraftMetadata(KRaftMetadataStorage.SHARED)
+                .build();
+        assertThat(VolumeUtils.kraftMetadataPath(markedSingleVolume), is("/var/lib/kafka/data"));
+
+        // Tests with unmarked JBOD storage
+        assertThat(VolumeUtils.kraftMetadataPath(JBOD_STORAGE), is("/var/lib/kafka/data-0"));
+
+        JbodStorage jbodStorageWithMixedOrder = new JbodStorageBuilder().withVolumes(
+                new PersistentClaimStorageBuilder()
+                        .withDeleteClaim(false)
+                        .withId(20)
+                        .withSize("20Gi")
+                        .build(),
+                new PersistentClaimStorageBuilder()
+                        .withDeleteClaim(true)
+                        .withId(1)
+                        .withSize("10Gi")
+                        .build(),
+                new PersistentClaimStorageBuilder()
+                        .withDeleteClaim(true)
+                        .withId(10)
+                        .withSize("10Gi")
+                        .build())
+                .build();
+        assertThat(VolumeUtils.kraftMetadataPath(jbodStorageWithMixedOrder), is("/var/lib/kafka/data-1"));
+
+        // Marked JBOD storage
+        JbodStorage markedJbodStorage = new JbodStorageBuilder().withVolumes(
+                new PersistentClaimStorageBuilder()
+                        .withDeleteClaim(false)
+                        .withId(0)
+                        .withSize("20Gi")
+                        .build(),
+                new PersistentClaimStorageBuilder()
+                        .withDeleteClaim(true)
+                        .withId(1)
+                        .withKraftMetadata(KRaftMetadataStorage.SHARED)
+                        .withSize("10Gi")
+                        .build())
+                .build();
+        assertThat(VolumeUtils.kraftMetadataPath(markedJbodStorage), is("/var/lib/kafka/data-1"));
+    }
```

Comment:
should we have a test about more than one volume marked for metadata storage?

Reply:
Test for what exactly? Multiple metadata volumes are checked in `StorageDiffTest.testMultipleKraftMetadataVolumes`. This is not supposed to be called with multiple metadata volumes.

Reply:
I missed that test in my pass then. Thanks.